src/c_api.cpp

```diff
@@ -268,10 +270,13 @@ class Booster {
   }
 
 
-  void Predict(int num_iteration, int predict_type, int nrow,
+  void Predict(int num_iteration, int predict_type, int nrow, int ncol,
                std::function<std::vector<std::pair<int, double>>(int row_idx)> get_row_fun,
                const Config& config,
                double* out_result, int64_t* out_len) {
+    if (ncol != boosting_->MaxFeatureIdx() + 1) {
```

Comment:
Doesn't it break the part about `0`?
https://github.com/microsoft/LightGBM/blob/1f88721a7982a804731b1dc108ee40625405639c/include/LightGBM/c_api.h#L686

Reply:
just realize we actually didn't guss the num_col when it is zero. 
I think it is better to pass non-zero values.